library/NetworkControl.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import ipaddress
+
+from library.common import *
+from urllib.parse import quote
+
+class NetworkControl(object):
+    def __init__(self, urlhandler, machine_id, interface_name):
+        self.urlhandler = urlhandler
+        self.machine_id = machine_id
+        self.interface = self.get_interface(interface_name)
+
+    def get_network_info(self, command):
+        if command == 'getip':
+            return self.get_ip()
+        elif command == 'getmac':
+            return self.get_mac()
+        elif command == 'getnetmask':
+            return self.get_netmask()
+        else:
+            raise LookupError("Unknown Command : %s" % command)
+
+    def get_interface(self, interface_name):
+        url = "/api/v1/machine/{}/interface".format(self.machine_id)
+        interfaces = self.urlhandler.get(url)
+
+        if not interfaces:
+            raise ProvisionerError("No interfaces for machine id %s" %
+                                   self.machine_id)
+
+        for i in interfaces:
+            if 'identifier' in i and str(i['identifier']) == interface_name:
+                return i
+            else:
```

Comment:
This forces the interfaces to be the first ones. You don't need the else, just let it fall to the last raise.